### DISCUSSION TITLE
App Router always renders twice (also in production)

### DISCUSSION THREAD

**Discussion by vaneenige** [vercel/next.js#58085] [App Router]
After creating a project with `npx create-next-app@latest` and selecting all default options I added a `console.log` in the Home component in `page.tsx`. Then I run `npm run build` and it is logged twice. **Is there a specific reason why this happens?**

It's always like this (regardless of things like React Strict Mode), also during revalidation and as mentioned, in production builds.

```
export default function Home() {
  console.log('home', Math.random());
```

```
Generating static pages (0/5)  [    ]
home 0.9315855929467511
home 0.18447993893122194
```

I couldn't find any issues or posts similar to this. It happens on Next 13 and Next 14.

**Comment by icyJoseph** [comment]
Isn't it related to this? https://github.com/vercel/next.js/discussions/52934#discussioncomment-6498597

**Reply by vaneenige** [reply]
That's exactly it, but I didn't come across it, thanks! I'm still interested to know **why** this actually happens though.
